### PR TITLE
Fix disabling spot instances when using launch templates

### PR DESCRIPTION
**Pull request by hakman**
Some extra fixes after #10187, to be able to remove the SpotPrice when already set.

/cc @rifelpet

### CODE REVIEW

File: upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go [hunk 1]
```diff
@@ -456,8 +456,8 @@ func (v *AutoscalingGroup) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *Autos
 
 		if changes.LaunchTemplate != nil {
 			spec := &autoscaling.LaunchTemplateSpecification{
-				LaunchTemplateId: changes.LaunchTemplate.ID,
-				Version:          &launchTemplateVersion,
+				LaunchTemplateName: changes.LaunchTemplate.ID,
```
Comment (rifelpet) [inline upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go:459]: I'm just curious, why is this change necessary?

Comment (hakman) [reply upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go:459]: There is some strange mix in the current code as you can see in other sections of the file:
https://github.com/kubernetes/kops/blob/b91a40c52aa67d10774d041754b74cc503513a44/upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go#L379

This will be changed in https://github.com/kubernetes/kops/pull/10151 where I tested the fix initially and this is why the current "mismatch".
